packages/config/src/handlers/application/applicationConfigHandler.ts

```diff
@@ -88,20 +97,15 @@ export async function updateApplicationConfigHandler(
   h: Hapi.ResponseToolkit
 ) {
   try {
-    const applicationConfig = request.payload as IApplicationConfigurationModel
-    const existingApplicationConfig: IApplicationConfigurationModel | null =
-      await ApplicationConfig.findOne({})
-    if (!existingApplicationConfig) {
-      throw badRequest('No existing application config found')
-    }
-    // Update existing application config fields
-    merge(existingApplicationConfig, applicationConfig)
-
-    await ApplicationConfig.update(
-      { _id: existingApplicationConfig._id },
-      existingApplicationConfig
+    const currentConfig = await getApplicationConfig()
+    const changeConfig = request.payload as IApplicationConfigurationModel
+    const applicationConfig = merge(currentConfig, changeConfig)
+    await ApplicationConfig.findOneAndUpdate(
+      {},
+      { $set: applicationConfig },
+      { upsert: true }
     )
-    return h.response(existingApplicationConfig).code(201)
+    return h.response(applicationConfig).code(201)
```

Comment:
We probably need to re-use the `getApplicationConfig` here as otherwise we aren't returning the total config from here